pyquil/quil.py

```diff
@@ -956,7 +931,7 @@ def __eq__(self, other: object) -> bool:
         return False
 
     def __len__(self) -> int:
-        return len(self.instructions)
+        return len(self._program.to_instructions())
```

Comment:
Why's this necessary, does `self._program.to_instructions()` also include the defgates? What do people typically use `len(Program)` for?

Reply:
Good catch, `len(Program)` should return the length of the `instructions` property for backwards compatibility. This change shouldn't have stuck around. It was a roundabout way of fixing an issue with the PyQVM that I've now addressed properly.